Doc/howto/a-conceptual-overview-of-asyncio.rst

```diff
@@ -0,0 +1,593 @@
+       ...
+
+In this example, imagine the event loop has passed control to the start of the
+coroutine ``plant_a_tree()``.
+As seen above, the coroutine creates a task and then awaits it.
+The ``await dig_the_hole_task`` instruction adds a callback (which will resume
+``plant_a_tree()``) to the ``dig_the_hole_task`` object's list of callbacks.
+And then, the instruction cedes control to the event loop.
+Some time later, the event loop will pass control to ``dig_the_hole_task``
+and the task will finish whatever it needs to do.
+Once the task finishes, it will add its various callbacks to the event loop,
+in this case, a call to resume ``plant_a_tree()``.
+
+Generally speaking, when the awaited task finishes (``dig_the_hole_task``),
+the original task or coroutine (``plant_a_tree()``) is added back to the event
+loops to-do list to be resumed.
+
+This is a basic, yet reliable mental model.
+In practice, the control handoffs are slightly more complex, but not by much.
+In part 2, we'll walk through the details that make this possible.
+
+**Unlike tasks, awaiting a coroutine does not hand control back to the event
+loop!**
+Wrapping a coroutine in a task first, then awaiting that would cede
+control.
+The behavior of ``await coroutine`` is effectively the same as invoking a
+regular, synchronous Python function.
+Consider this program::
+
+   import asyncio
+
+   async def coro_a():
+      print("I am coro_a(). Hi!")
+
+   async def coro_b():
+      print("I am coro_b(). I sure hope no one hogs the event loop...")
+
+   async def main():
+      task_b = asyncio.create_task(coro_b())
+      num_repeats = 3
+      for _ in range(num_repeats):
+         await coro_a()
+      await task_b
+
+   asyncio.run(main())
+
+The first statement in the coroutine ``main()`` creates ``task_b`` and schedules
+it for execution via the event loop.
+Then, ``coro_a()`` is repeatedly awaited. Control never cedes to the
+event loop which is why we see the output of all three ``coro_a()``
+invocations before ``coro_b()``'s output:
+
+.. code-block:: none
+
+   I am coro_a(). Hi!
+   I am coro_a(). Hi!
+   I am coro_a(). Hi!
+   I am coro_b(). I sure hope no one hogs the event loop...
+
+If we change ``await coro_a()`` to ``await asyncio.create_task(coro_a())``, the
+behavior changes.
+The coroutine ``main()`` cedes control to the event loop with that statement.
+The event loop then proceeds through its backlog of work, calling ``coro_b()``
+and then ``coro_a()`` before resuming the coroutine ``main()``.
+
+.. code-block:: none
+
+   I am coro_b(). I sure hope no one hogs the event loop...
+   I am coro_a(). Hi!
+   I am coro_a(). Hi!
+   I am coro_a(). Hi!
+
+This behavior of ``await coroutine`` can trip a lot of people up!
+That example highlights how using only ``await coroutine`` could
+unintentionally hog control from other tasks and effectively stall the event
+loop.
+:func:`asyncio.run` can help you detect such occurences with the ``debug=True``
```

Comment:
To clarify, the way I'd expect most developers to use this is not by modifying the code, but by running python in dev mode locally (`-X dev` or `PYTHONDEVMODE=1`). A link should take users to: https://docs.python.org/3/library/asyncio-dev.html#asyncio-debug-mode